File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n);
        int s = 0, e = n - 1;
        for(int i = n - 1; i >= 0; i--)
        {
            if(abs(nums[s]) >= abs(nums[e]))
            {
                res[i] = nums[s] * nums[s];
                s++;
            }
            else
            {
                res[i] = nums[e] * nums[e];
                e--;
            }
        }

        return res;
    }
};

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution(object):
    def sortedSquares(self, nums):
        n = len(nums)
        res = [0] * n
        i = 0 
        j = n - 1
        s = n - 1
        while s >= 0:
            if abs(nums[i]) < abs(nums[j]):
                res[s] = nums[j] * nums[j]
                j -= 1
                s -= 1
            else:
                res[s] = nums[i] * nums[i]
                i += 1
                s -= 1
        
        return res
        